### ISSUE TITLE
Automatically run tournaments nightly

### ISSUE THREAD

**Issue by marcharper** [Axelrod-Python/tournament#7]
We discussed this once upon a time. The update script now takes over an hour to run (on a fast machine), and of course is going to get longer as more strategies are added to the library.

I'm thinking it's a good time to add a script to pull down the latest axelrod master, checkout a remote tracking branch, install it, and run the update_script.

My questions are:
(1) Where will it run? I can do it locally for now... it might be a good idea to run in more than one location.
(2) What to do with the new figures? I can try to automatically push to my tournament repository, which can then be merged if anyone needs to do so.

**Comment by drvinceknight** [comment]
> (1) Where will it run? I can do it locally for now... it might be a good idea to run in more than one location.

I could run it locally also. There's a fair few machines at the university that would be ok for this to run on.

Also, is this something travis could do for us, not sure what the timeout limit is?

> (2) What to do with the new figures? I can try to automatically push to my tournament repository, which can then be merged if anyone needs to do so.

I'm not sure I understand, why not just push to the tournament repository?

Also, I'm guessing you don't mean for this to actually run nightly? We could just check the diff and run if new strategies are present?

**Comment by marcharper** [comment]
> Also, is this something travis could do for us, not sure what the timeout limit is?

I think travis has a ~10 minute time limit, which is not long enough.

> Also, I'm guessing you don't mean for this to actually run nightly? We could just check the diff and run if new strategies are present?

Ideally, the script would check if there are new commits to master before running, but I'm not sure how easy that is to do.

> I'm not sure I understand, why not just push to the tournament repository?

We could push it to a side branch on the main repository, that's fine, but I'm not sure we want to merge /push to gh_pages automatically (not sure if that's what you meant).

**Comment by drvinceknight** [comment]
> We could push it to a side branch on the main repository, that's fine, but I'm not sure we want to merge /push to gh_pages automatically (not sure if that's what you meant).

Ah yes: very good point. I like the other branch idea, we could use the github API to open a PR for one of us to review?

**Comment by marcharper** [comment]
Sounds good.

**Comment by drvinceknight** [comment]
We could also do something like this https://github.com/tdhopper/phd-tweet and have a twitter account shout when there are new strategies...

**Comment by drvinceknight** [comment]
Given the size of this and the fact that data is now being put on zenodo I think this needs to be done manually. Ok to close?

**Comment by marcharper** [comment]
It's up to you -- I don't mind running a cron job every night if there's interest.

**Comment by drvinceknight** [comment]
> It's up to you -- I don't mind running a cron job every night if there's interest.

The main thing would be to make sure we're notified so that we can upload the data files... Fancy hooking in to the github API so it would automatically open a PR? It probably makes sense (for computing time) for me to run it on our cluster but if you're happy to: :+1:

**Comment by marcharper** [comment]
Yeah either way. I was just going to dump the data to dropbox or my website and I don't think it would be difficult to automatically open a PR. But I also doubt we want to merge that often. It would just be nice to have some recent runs of data laying around somewhere easily accessible.

**Comment by drvinceknight** [comment]
Cool. As you say, it's not exactly a problem if we're running the tournament more than once :)

I was talking about updating the data here: https://zenodo.org/record/49557#.VxlIgFQrKkA It's a shame that that can't just update, it's (mildly) tedious to update as you have to create a new record each time.

Let me know if you want me to run it :)
